docs/topics/whatsnew-eap.md

```diff
@@ -269,9 +562,173 @@ There are new Gradle properties so that you can configure a directory of your ch
+
+The Kotlin/Wasm compiler collects any [top-level functions](wasm-js-interop.md#functions-with-the-jsexport-annotation) 
+marked with `@JsExport` and automatically generates TypeScript definitions in a `.d.ts` file.
+
+To generate TypeScript definitions, in your `build.gradle.kts` file in the wasmJs section, 
+add the `generateTypeScriptDefinitions()` function:
+
+```kotlin
+kotlin {
+    wasmJs {
+        binaries.executable()
+        browser {
+        }
+        generateTypeScriptDefinitions()
+    }
+}
+```
+
+## Kotlin/JS
+
+This version brings the following changes:
+* [Support for type-safe plain JavaScript objects](#support-for-type-safe-plain-javascript-objects)
+* [Support for npm package manager](#support-for-npm-package-manager)
+ 
+### Support for type-safe plain JavaScript objects
+
+> The `js-plain-objects` plugin is [Experimental](components-stability.md#stability-levels-explained).
+> It may be dropped or changed at any time. The `js-plain-objects` plugin **only** supports the K2 compiler.
+>
+{type="warning"}
+
+To make it easier to work with JavaScript APIs, in Kotlin 2.0.0-Beta4 we provide a new plugin: 
+[`js-plain-objects`](https://github.com/JetBrains/kotlin/tree/master/plugins/js-plain-objects),
+that you can use to create type-safe plain JavaScript objects.
+The plugin checks your code for any [external interfaces](wasm-js-interop.md#external-interfaces) 
+that have a `@JsPlainObject` annotation and adds:
+
+* an inline `invoke` operator function inside the companion object that you can use as a constructor.
+* a `.copy()` function that you can use to create a copy of your object while adjusting some of its properties.
+
+For example:
+
+```kotlin
+import kotlinx.js.JsPlainObject
+
+@JsPlainObject
+external interface User {
+    var name: String
+    val age: Int
+    val email: String? 
+}
+
+fun main() {
+    // Creates a JavaScript object
+    val user = User(name = "Name", age = 10)
+    // Copies the object and adds an email
+    val copy = user.copy(age = 11, email = "some@user.com")
+
+    println(JSON.stringify(user)) 
+    // { "name": "Name", "age": 10 }
+    println(JSON.stringify(copy)) 
+    // { "name": "Name", "age": 11, "email": "some@user.com" }
+}
+```
+
+Any JavaScript objects created with this approach are safer because instead of only seeing errors at runtime, 
+you can see them at compile time, or even highlighted by your IDE.
+
+Consider this example that uses a `fetch()` function to interact with a JavaScript API, using external interfaces 
+to describe the shape of the JavaScript objects:
+
+```kotlin
+import kotlinx.js.JsPlainObject
+
+@JsPlainObject
+external interface FetchOptions {
+    val body: String?
+    val method: String
+}
+
+// A wrapper for Window.fetch
+suspend fun fetch(url: String, options: FetchOptions? = null) = TODO("Add your custom behavior here")
+
+// A compile-time error is triggered as metod is not recognized
```

Comment:
```suggestion
// A compile-time error is triggered as the method is not recognized
```

Reply:
The use of `metod` is intentional but hard to show in code blocks without inline code formatting. I added some extra text to try and make it clear that `metod` is not recognized as `method`. The use of `metod` was undetected before but is now reported as an error with this change. I will mark other comments related to this as resolved.